django/contrib/auth/views.py

```diff
@@ -334,6 +346,117 @@ def password_reset_complete(request,
+    form_class = PasswordResetForm
+    success_url = reverse_lazy('password_reset_done')
+
+    title = _('Password reset')
+    email_template_name = 'registration/password_reset_email.html'
+    html_email_template_name = None
+    extra_email_context = None
+    subject_template_name = 'registration/password_reset_subject.txt'
+    from_email = None
+    token_generator = default_token_generator
+
+    @method_decorator(csrf_protect)
+    def dispatch(self, *args, **kwargs):
+        return super(PasswordResetView, self).dispatch(*args, **kwargs)
+
+    def form_valid(self, form):
+        opts = {
+            'use_https': self.request.is_secure(),
+            'token_generator': self.token_generator,
+            'from_email': self.from_email,
+            'email_template_name': self.email_template_name,
+            'subject_template_name': self.subject_template_name,
+            'request': self.request,
+            'html_email_template_name': self.html_email_template_name,
+            'extra_email_context': self.extra_email_context,
+        }
+        form.save(**opts)
+        return super(PasswordResetView, self).form_valid(form)
+
+
+class PasswordResetDoneView(PasswordContextMixin, TemplateView):
+    template_name = 'registration/password_reset_done.html'
+    title = _('Password reset sent')
+
+
+class PasswordResetConfirmView(PasswordContextMixin, FormView):
+    template_name = 'registration/password_reset_confirm.html'
+    form_class = SetPasswordForm
+    success_url = reverse_lazy('password_reset_complete')
+    title = _('Enter new password')
+    token_generator = default_token_generator
+
+    @method_decorator(sensitive_post_parameters())
+    @method_decorator(never_cache)
+    def dispatch(self, *args, **kwargs):
+        assert 'uidb64' in kwargs and 'token' in kwargs
+        return super(PasswordResetConfirmView, self).dispatch(*args, **kwargs)
+
+    def get_user(self, uidb64):
+        UserModel = get_user_model()
+        try:
+            # urlsafe_base64_decode() decodes to bytestring on Python 3
+            uid = force_text(urlsafe_base64_decode(uidb64))
+            user = UserModel._default_manager.get(pk=uid)
+        except (TypeError, ValueError, OverflowError, UserModel.DoesNotExist):
+            user = None
+        return user
+
+    def get_form_kwargs(self):
+        kwargs = super(PasswordResetConfirmView, self).get_form_kwargs()
+        kwargs['user'] = self.get_user(self.kwargs['uidb64'])
+        return kwargs
+
+    def form_valid(self, form):
+        form.save()
+        return super(PasswordResetConfirmView, self).form_valid(form)
+
+    def get_context_data(self, **kwargs):
+        context = super(PasswordResetConfirmView, self).get_context_data(**kwargs)
+        user = context['form'].user
+        if user is not None and self.token_generator.check_token(user, self.kwargs['token']):
+            context['validlink'] = True
+        else:
+            del context['form']
```

Comment:
Not sure it makes a difference but before it looks like we got `form=None` in the context.